### PR TITLE
Created a  new Object - Thermals - Specifications to capture temperature and thermal resistance data

### PR DESCRIPTION
**Pull request by asadate**
Created a  new Object , Thermals -  Specifications to capture temperature and thermal resistance data

### CODE REVIEW

File: part-spec/common/thermals.json [hunk 1]
```diff
@@ -0,0 +1,81 @@
+{
+    "$id": "https://github.com/edatasheets/digital-datasheets/blob/main/part-spec/common/thermals.json",
+    "$schema": "http://json-schema.org/draft-07/schema#",
+    "title": "specification of thermals ",
```
Comment (rsnancollas) [inline part-spec/common/thermals.json]: specification of thermal properties of components

File: part-spec/common/thermals.json [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+    "thermalSpec": {
```
Comment (rsnancollas) [inline part-spec/common/thermals.json]: maybe 

thermalProperties

File: part-spec/common/thermals.json [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+        "type": "object",
+        "properties": {
+            "junctionTemp": {
+                "description": "recommended operating junction temperature range of a part",
+                "comment": "units of Celsius",
+                "$ref": "../common/values.json#/valueOptions"
+            },
+            "junctionTempMax": {
+                "description": "absolute maximum junction temperature range of a part",
```
Comment (rsnancollas) [inline part-spec/common/thermals.json]: is this a range or just a single temperature?

File: part-spec/common/thermals.json [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+                "comment": "units of Celsius",
+                "$ref": "../common/values.json#/valueOptions"
+            },
+            "ambientTemp": {
```
Comment (rsnancollas) [inline part-spec/common/thermals.json]: General comment - do we want to use Temp or Temperature?

File: part-spec/common/thermals.json [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+                "description": "recommended operating ambient air temperature range of a part",
+                "comment": "units of Celsius",
+                "$ref": "../common/values.json#/valueOptions"
+            },
+            "ambientTempMax": {
+                "description": "absolute maximum ambient air temperature range under which a part operates",
```
Comment (rsnancollas) [inline part-spec/common/thermals.json]: maybe: "... operates as expected"

File: part-spec/common/thermals.json [hunk 6]
```diff
@@ -0,0 +1,81 @@
...
+                "comment": "units of Celsius",
+                "$ref": "../common/values.json#/valueOptions"
+            },
+            "leadTempMax": {
+                "description": "absolute maximum lead temperature of a part",
+                "comment": "units of Celsius",
+                "$ref": "../common/values.json#/valueOptions"
+            },
+            "storageTempMax": {
+                "description": "absolute maximum storage temperature range of a part",
```
Comment (rsnancollas) [inline part-spec/common/thermals.json]: Sometimes you've used range for max and sometimes you just used temperature. I think we should pick one and be consistent.

I think "range" is fine for any of the non-max temperatures.

File: part-spec/common/thermals.json [hunk 7]
```diff
@@ -0,0 +1,81 @@
...
+                "comment": "units of Celsius",
+                "$ref": "../common/values.json#/valueOptions"
+            },
+            "packageThermalResistance": {
+                "description": "package thermal resistance of a part",
+                "comment": "units of Celsius/watt or kelvin/watt",
+                "$ref": "../common/values.json#/valueOptions"
+            },
+            "ThermalResistanceJunctionToAmbient": {
```
Comment (rsnancollas) [inline part-spec/common/thermals.json]: To be consistent with other files, I think we should use lower case letters to start all the property names

File: part-spec/common/thermals.json [hunk 8]
```diff
@@ -0,0 +1,81 @@
+{
+    "$id": "https://github.com/edatasheets/digital-datasheets/blob/main/part-spec/common/thermals.json",
```
Comment (rsnancollas) [inline part-spec/common/thermals.json:2]: One thing that might be worthwhile is getting this reviewed by a power/thermal engineer - I've worked with a few folks at Intel over the years, not sure if they would be willing to give a quick read-over?

File: part-spec/common/thermals.json [hunk 9]
```diff
@@ -0,0 +1,81 @@
...
+                "description": "package thermal resistance of a part",
+                "comment": "units of Celsius/watt or kelvin/watt",
+                "$ref": "../common/values.json#/valueOptions"
+            },
+            "ThermalResistanceJunctionToAmbient": {
+                "description": "thermal resistance between the semiconductor junction of a part and the ambient air",
+                "comment": "units of Celsius/watt or kelvin/watt",
+                "$ref": "../common/values.json#/valueOptions"
+            },
+            "ThermalResistanceJunctionToCase": {
```
Comment (rsnancollas) [inline part-spec/common/thermals.json]: lowercase

File: part-spec/common/thermals.json [hunk 10]
```diff
@@ -0,0 +1,81 @@
...
+                "description": "thermal resistance between the semiconductor junction and the package surface of a part",
+                "comment": "units of Celsius/watt or kelvin/watt",
+                "$ref": "../common/values.json#/valueOptions"
+            },
+            "ThermalResistanceJunctionToboard": {
```
Comment (rsnancollas) [inline part-spec/common/thermals.json]: lowercase

File: part-spec/common/thermals.json [hunk 11]
```diff
@@ -0,0 +1,81 @@
...
+                "description": "thermal resistance between the semiconduction junction of a part and the board",
+                "comment": "units of Celsius/watt or kelvin/watt",
+                "$ref": "../common/values.json#/valueOptions"
+            },
+            "ThermalResistanceJunctionToLead": {
```
Comment (rsnancollas) [inline part-spec/common/thermals.json]: lowercase

File: part-spec/common/thermals.json [hunk 12]
```diff
@@ -0,0 +1,81 @@
...
+                "description": "thermal resistance between the semiconduction junction of a part and the soldering point",
+                "comment": "units of Celsius/watt or kelvin/watt",
+                "$ref": "../common/values.json#/valueOptions"
+            },
+            "ThermalResistanceCaseToAmbient": {
```
Comment (rsnancollas) [inline part-spec/common/thermals.json]: lowercase

File: part-spec/common/thermals.json [hunk 13]
```diff
@@ -0,0 +1,81 @@
...
+                "comment": "units of Celsius/watt or kelvin/watt",
+                "$ref": "../common/values.json#/valueOptions"
+            },
+            "ThermalResistanceJunctionToboard": {
+                "description": "thermal resistance between the semiconduction junction of a part and the board",
+                "comment": "units of Celsius/watt or kelvin/watt",
+                "$ref": "../common/values.json#/valueOptions"
+            },
+            "ThermalResistanceJunctionToLead": {
+                "description": "thermal resistance between the semiconduction junction of a part and the soldering point",
```
Comment (rsnancollas) [inline part-spec/common/thermals.json]: how about: "...junction of a part and the solder pad"

File: part-spec/component.json [hunk 14]
```diff
@@ -26,6 +26,10 @@
 			"description": "component package information",
 			"$ref": "./common/package.json#/package"
 		},
+		"thermal": {
+			"description": "component temperature and thermalresistance information",
```
Comment (rsnancollas) [inline part-spec/component.json]: thermalresistance -> thermal resistance

File: part-spec/common/thermals.json (hunk 8 above)
Comment (asadate) [reply part-spec/common/thermals.json:2]: Good idea.

File: part-spec/common/thermals.json (hunk 1 above)
Comment (asadate) [reply part-spec/common/thermals.json]: done

File: part-spec/common/thermals.json (hunk 2 above)
Comment (asadate) [reply part-spec/common/thermals.json]: changed

File: part-spec/common/thermals.json (hunk 3 above)
Comment (asadate) [reply part-spec/common/thermals.json]: It is a range

File: part-spec/common/thermals.json (hunk 4 above)
Comment (asadate) [reply part-spec/common/thermals.json]: I was thinking the names would be too long but we should probably use temperature to remove any confusion.

File: part-spec/common/thermals.json (hunk 6 above)
Comment (asadate) [reply part-spec/common/thermals.json]: I'm using Max to differentiate  the abs Max values from the recommended operating range values. In the datasheets I've seen some of the Max values are consistently expressed as a range (Min and Max) while other are expressed as a single Max value.  To simplify, I've changed names to add AbsMax and remove the word "range" from the definition.

File: part-spec/common/thermals.json (hunk 7 above)
Comment (asadate) [reply part-spec/common/thermals.json]: Yes. This is a typo.

File: part-spec/common/thermals.json (hunk 13 above)
Comment (asadate) [reply part-spec/common/thermals.json]: changed

File: part-spec/component.json (hunk 14 above)
Comment (asadate) [reply part-spec/component.json]: changed

File: part-spec/common/thermals.json (hunk 5 above)
Comment (asadate) [reply part-spec/common/thermals.json]: This is abs max, so more like values beyond which the part should not be operated. I've changed the wording to keep it simpler, hopefully.

**Review by rsnancollas — APPROVED** [review]

**Pull request merged**